### PR TITLE
[FIX] sql_request_abstract: regression

### PR DESCRIPTION
**Pull request by thomaspaulb**
Fixes regression introduced by [this commit](https://github.com/OCA/server-tools/commit/f0267eac7ffdd9bb13d9bad6e71c8cd0fe654cec) leading to issues with tests such as:

```
    rollback_name = self._create_savepoint()
TypeError: _create_savepoint() missing 1 required positional argument: 'cr'
```

@florian-dacosta

### CODE REVIEW

File: sql_request_abstract/models/sql_request_mixin.py [hunk 1]
```diff
@@ -296,7 +296,7 @@ def _check_execution(self):
         is done after."""
         self.ensure_one()
         query = self._prepare_request_check_execution()
-        rollback_name = self._create_savepoint()
+        rollback_name = self._create_savepoint(self.env.cr)
```
Comment (florian-dacosta) [inline sql_request_abstract/models/sql_request_mixin.py]: Thanks for the fix @thomaspaulb 
To be consistent with last introduced feature, it think it should be : 
```
query_cr = self._get_cr_for_query()
rollback_name = self._create_savepoint(query_cr)
```

Comment (thomaspaulb) [reply sql_request_abstract/models/sql_request_mixin.py]: @florian-dacosta And then only use `query_cr` throughout the remainder of the function?

Comment (thomaspaulb) [reply sql_request_abstract/models/sql_request_mixin.py]: Pushed it that way, please check.

**Review by florian-dacosta — APPROVED** [review]

**Comment by florian-dacosta** [comment]
@legalsylvain

**Review by misern2 — APPROVED** [review]

**Comment by thomaspaulb** [comment]
Squashed. @StefanRijnhart are you someone who can merge?

**Comment by thomaspaulb** [comment]
@hbrunn

**Comment by thomaspaulb** [comment]
Thanks @StefanRijnhart

**Pull request merged**